docs/source/en/conceptual/reproducibility.mdx

```diff
@@ -0,0 +1,159 @@
+[PyTorch's statement about reproducibility](https://pytorch.org/docs/stable/notes/randomness.html).
+
+PyTorch states that 
+> *completely reproducible results are not guaranteed across PyTorch releases, individual commits, or different platforms.*
+While one can never expect the same results across platforms, one can expect results to be reproducible 
+across releases, platforms, etc... within a certain tolerance. However, this tolerance strongly varies 
+depending on the diffusion pipeline and checkpoint.
+
+In the following, we should how to best control sources of randomness for diffusion models.
+
+## Inference
+
+During inference, diffusion pipelines heavily rely on random sampling operations, such as the creating the 
+gaussian noise tensors to be denoised as well as adding noise to the scheduling step.
+
+Let's have a look at an example. We run the [DDIM pipeline](./api/pipelines/ddim.mdx) 
+for just two inference steps and return a numpy tensor to look into the numerical values of the output.
+
+```python
+from diffusers import DDIMPipeline
+import numpy as np
+
+model_id = "google/ddpm-cifar10-32"
+
+# load model and scheduler
+ddim = DDIMPipeline.from_pretrained(model_id)
+
+# run pipeline for just two steps and return numpy tensor
+image = ddim(num_inference_steps=2, output_type="np").images
+print(np.abs(image).sum())
+```
+
+Running the above prints a value of 1464.2076, but running it again prints a different 
+value of 1495.1768. What is going on here? Every time the pipeline is run, gaussian noise 
+is created and step-wise denoised. To create the gaussian noise with [`torch.randn`](https://pytorch.org/docs/stable/generated/torch.randn.html) a different random seed is taken every time, thus leading to a different result.
+This is a desired property of diffusion pipelines as it means that the pipeline can create a different random image every time it is run. In many cases, one would like to generate the exact same image of a certain 
+run, for which case an instance of a [PyTorch generator](https://pytorch.org/docs/stable/generated/torch.randn.html) has to be passed:
+
+```python
+import torch
+from diffusers import DDIMPipeline
+import numpy as np
+
+model_id = "google/ddpm-cifar10-32"
+
+# load model and scheduler
+ddim = DDIMPipeline.from_pretrained(model_id)
+
+# create a generator for reproducibility
+generator = torch.Generator(device="cpu").manual_seed(0)
+
+# run pipeline for just two steps and return numpy tensor
+image = ddim(num_inference_steps=2, output_type="np", generator=generator).images
+print(np.abs(image).sum())
+```
+
+Running the above always prints a value of 1491.1711 - also upon running it again because we 
+define the generator object to be passed to all random functions of the pipeline.
+
+If you run this code snippet on your specific hardware and version, you should get a very similar, if not the same result.
```

Comment:
```suggestion
Running the above always prints a value of 1491.1711 - also upon running it again because we 
define the generator object to be passed to all random functions of the pipeline.

If you run this code snippet on your specific hardware and version, you should get a similar, if not the same, result.
```